pkg/controller/kfservice/kfservice_controller.go

```diff
@@ -117,6 +110,11 @@ type ReconcileService struct {
 	Recorder record.EventRecorder
 }
 
+// Reconciler is implemented by all subresources
+type Reconciler interface {
```

Comment:
The controller runtime also defines an interface named Reconciler, maybe rename this SubReconciler or something?

Reply:
I honestly think this is fine as is. It's the same concept.